### ISSUE TITLE
Bug: [consistent-type-assertions] false negative with `satifies` and `as`

### ISSUE THREAD

**Issue by FloEdelmann** [typescript-eslint/typescript-eslint#10827]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.5.2&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgGIHt3IN4ChnIBGcUAXMgM5hSgDm%2BRcAXuVTSPQL664LohVkMTAEZkAXhwNiZZAHI5uTpThhgFGMAgVkABRJq4AGwA8GdAD5kcHeZ58BYIZgBMEqQRnkFS67cxAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWgENkyRNHyUuqDAG1w2HFOgdokADTKVwKBwBGAK0Rl8AGUr9oo%2BABUSiAIISpMueihNEANymQYAC%2BumAAusohQUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
interface Foo {
  bar: string
  baz: string
}

const foo1 = {
  bar: ''
} satisfies Partial<Foo> as Foo

const foo2 = {
  bar: ''
} as Foo
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/consistent-type-assertions": [
      "error",
      { objectLiteralTypeAssertions: "never" }
    ]
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Warning for both `foo1` and `foo2`, since `as` is used in both.

### Actual Result

Only warning for `foo2`. No warning for `foo1`, which also uses `satisfies`.

**Comment by bradzacher** [comment]
I'm curious as to why your code is doing `satisfies` followed by `as`. It doesn't really add any safety for object literals - TS performs the same checks on a bare `as`. [playground](https://typescript-eslint.io/play/#ts=5.5.2&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgGIHt3IN4ChnIBGcUAXMgM5hSgDm%2BRcAXuVTSPQL664LohVkMTAEZkAXhwNiZZAHI5AGmQAPNeRG5OlOGGAUYwCBWQAFEnrgAbADwZ0APmRwT9nnwFghmAEwSpBDLkCspqKhpazq6YuEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWgENkyRNHyUuqDAG1w2HBwBmayABplK4FA4AjAFaIy%2BADKV%2B0UfAAqJRAEEJUmXPRQmiAG5TIMABfXTAAXWUQoKA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by FloEdelmann** [comment]
I'm using a library function that accepts a `Foo` parameter, but its implementation would work with a `Partial<Foo>` argument as well. So I make sure that my object satisfies the `Partial<Foo>` constraint, but then cast it to `Foo` to be able to pass it to that function.

**Comment by kirkwaiblinger** [comment]
> I'm using a library function that accepts a `Foo` parameter, but its implementation would work with a `Partial<Foo>` argument as well. So I make sure that my object satisfies the `Partial<Foo>` constraint, but then cast it to `Foo` to be able to pass it to that function.

So, that makes sense as why you'd use it - now, what is the motivation for banning it?

**Comment by FloEdelmann** [comment]
That's a good question. I honestly just thought "the rule was written before `satisfies` existed, so it must be a mistake that it doesn't warn here".

I've opened a PR at the library in question that would allow `Partial<Foo>` parameters if accepted. Then `satisfies Partial<Foo> as Foo` wouldn't be necessary anymore and I could just write `const foo: Partial<Foo> = …` instead. But `satisfies Partial<Foo> as Foo` would still be correct, so neither TypeScript nor typescript-eslint would nudge me towards simplifying the construct.

Maybe if `consistent-type-assertions` would only be set to `warn` and I wouldn't disable it, then after the library is updated I'd see the warning at some point in the future and simplify it then. But if it is set to `error` and I need to disable it now, we're back to square one.

I actually don't know what behavior I actually want 😅

**Comment by JoshuaKGoldberg** [comment]
Sounds like we're good to close this issue as non-actionable on our end. Someone please yell at me if I'm wrong on that. 🙂
